core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -161,6 +161,13 @@ public void onNext(HttpObject o) {
                     }
                     merged = headers;
                 } else {
+                    if (responseEncoder.isResponseHeadersSent(req.id(), req.streamId())) {
+                        // The response is sent by the HttpRequestDecoder so we just cancel the stream message.
+                        isComplete = true;
+                        setDone(true);
+                        return;
+                    }
```

Comment:
Is it better to reject by `ServerHttp*ObjectEncoder.doWriteHeaders()` when a response headers tries to write after another response headers has been written? 🤔

Reply:
I think we should reject it here because otherwise, we log the responseheaders which is not sent actually. 🤔

Reply:
I see. This seems like the best we can do now.